### PR TITLE
SNAP-584

### PR DESCRIPTION
**Pull request by dshirish**
Modified snappy-tools/sbin/snappy-start-all.sh not to start lead node when rowstore arg is passed as discussed earlier.

Added a test in QueryRoutingDUnitTest.scala to confirm that GemXD URL works.

[Update]: As discussed updated the scripts to use GfxdLauncher when started in rowstore mode.

@kneeraj @sumwale please take a look.

### CODE REVIEW

File: snappy-tools/sbin/snappy-start-all.sh [hunk 1]
```diff
@@ -34,4 +34,6 @@ sbin="`cd "$sbin"; pwd`"
 "$sbin"/snappy-servers.sh start
 
 # Start Leads
-"$sbin"/snappy-leads.sh start
+if [ "$1" != "rowstore" ]; then
+  "$sbin"/snappy-leads.sh start
```
Comment (kneeraj) [inline snappy-tools/sbin/snappy-start-all.sh]: For stopping if we maintain that the cluster was started with rowstore option then skip stopping the lead node.

**Comment by kneeraj** [comment]
Looks good to me.

**Comment by dshirish** [comment]
@sumwale 
As discussed updated the scripts to use GfxdLauncher when started in rowstore mode.
Please review.

File: snappy-tools/bin/snappy-shell [hunk 2]
```diff
@@ -27,4 +27,19 @@ then
...
+
+function setnewargs() { newargs="$@"; }
+
+if echo $@ | grep -qw "rowstore"; then
+  #using rowstore launcher
+  newargs=
+  for arg in "$@"; do
+    if [[ "$arg" != "rowstore" ]] ; then
+      setnewargs "$newargs" "$arg"
+    fi
```
Comment (sumwale) [inline snappy-tools/bin/snappy-shell]: Won't this be better as "-rowstore"? Latter is our normal convention for other options while "start/stop/status" denote actions to be done.

File: snappy-tools/sbin/snappy-start-all.sh (hunk 1 above)
Comment (sumwale) [inline snappy-tools/sbin/snappy-start-all.sh]: See above comment about "-rowstore"

**Comment by sumwale** [comment]
Looks good. I had added comment about "-rowstore" but removed it since "snappy-shell rowstore start" also seems fine.

**Pull request merged**